compiler/semantic/op.go

```diff
@@ -212,6 +232,22 @@ func (a *analyzer) semPool(p *ast.Pool) ([]dag.Op, error) {
 	return sources, nil
 }
 
+func (a *analyzer) strConstOrName(name string) (string, error) {
+	e, err := a.scope.LookupExpr(name)
+	if err != nil || e == nil {
+		return name, err
+	}
+	l, ok := e.(*dag.Literal)
+	if !ok {
+		return "", fmt.Errorf("%s: string value required", name)
+	}
+	v := zson.MustParseValue(a.zctx, l.Value)
```

Comment:
zson.MustParseValue can panic. Use zson.ParseValue and handle the error instead.

And a nit: use val for zed.Values and save v for native Go values.
```suggestion
	val, err := zson.ParseValue(a.zctx, l.Value)
	if err != nil {
		return "", err
	}
```

Reply:
I think this panic would be desirable since this is merely re-parsing a value that was already checked by const. If there's an error here then something is seriously wrong.